provisioning/service/src/Authentication/SecurityHelper.cs

```diff
@@ -2,14 +2,17 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
 using System.Globalization;
 
 namespace Microsoft.Azure.Devices
 {
     internal static class SecurityHelper
     {
-        public static void ValidateServiceHostName(string serviceHostName, string serviceName)
+        internal static void ValidateServiceHostName(string serviceHostName, string serviceName)
         {
+            Debug.Assert(!string.IsNullOrWhiteSpace(serviceHostName));
+            Debug.Assert(!string.IsNullOrWhiteSpace(serviceName));
```

Comment:
These will need some comments otherwise it won't be clear why it failed. Comment is the second parameter.